# M3-B1 — Exploration des 3 sources Acerox

> Notebook d'**exploration rapide** — pas d'EDA fouillée, juste assez pour
> remplir l'inventaire de la note d'identification.

Auteur·rice : `Joelle Germanos` — Date : `30-06-2026`

**Règles** :
- Pas de transformation (juste lecture, `info`, `head`, `describe`)
- Une cellule markdown par source — qu'est-ce que tu observes ?
- Trace les **risques** et **questions** qui émergent pour l'`identification_sources.md`

In [3]:
from pathlib import Path
import json

import pandas as pd

DATA_DIR = Path("../data")

## Source 1 — Capteurs IoT (CSV)

In [4]:
df_iot = pd.read_csv(DATA_DIR / "capteurs_iot.csv")
df_iot.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51000 entries, 0 to 50999
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   timestamp      51000 non-null  object 
 1   site           51000 non-null  object 
 2   line_id        51000 non-null  int64  
 3   sensor_id      51000 non-null  object 
 4   temperature_c  51000 non-null  float64
 5   vibration_mms  50251 non-null  float64
 6   debit_uh       51000 non-null  float64
dtypes: float64(3), int64(1), object(3)
memory usage: 2.7+ MB


In [ ]:
# TODO — describe, head, value_counts par site/line, manquants

In [16]:
print(f"Volume : {len(df_iot):,} lignes × {len(df_iot.columns)} colonnes")
print(f"Période : {df_iot['timestamp'].min()} → {df_iot['timestamp'].max()}")
df_iot.info()
df_iot.describe()
print(df_iot.isna().sum())
print(df_iot['site'].value_counts())

Volume : 51,000 lignes × 7 colonnes
Période : 2026-04-01T00:00:38 → 2026-04-29T23:57:53
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51000 entries, 0 to 50999
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   timestamp      51000 non-null  object 
 1   site           51000 non-null  object 
 2   line_id        51000 non-null  int64  
 3   sensor_id      51000 non-null  object 
 4   temperature_c  51000 non-null  float64
 5   vibration_mms  50251 non-null  float64
 6   debit_uh       51000 non-null  float64
dtypes: float64(3), int64(1), object(3)
memory usage: 2.7+ MB
timestamp          0
site               0
line_id            0
sensor_id          0
temperature_c      0
vibration_mms    749
debit_uh           0
dtype: int64
site
Roubaix          20570
Saint-Etienne    20248
Lyon             10182
Name: count, dtype: int64


In [6]:
import pandas as pd

# Aperçu des premières lignes (déjà en tableau ✔)
display(df_iot.head())

# Statistiques descriptives (déjà en tableau ✔)
display(df_iot.describe())

# ----------- Tableau synthétique global -----------

summary_df = pd.DataFrame({
    "Metric": ["Volume lignes", "Nombre colonnes", "Période début", "Période fin"],
    "Valeur": [
        len(df_iot),
        len(df_iot.columns),
        df_iot['timestamp'].min(),
        df_iot['timestamp'].max()
    ]
})

display(summary_df)


# ----------- Valeurs manquantes -----------

missing_df = df_iot.isna().sum().reset_index()
missing_df.columns = ["Colonne", "Nb valeurs manquantes"]
display(missing_df)


# ----------- Répartition par site -----------

site_df = df_iot["site"].value_counts().reset_index()
site_df.columns = ["Site", "Nombre de lignes"]
display(site_df)


# ----------- Répartition par ligne -----------

line_df = df_iot["line_id"].value_counts().reset_index()
line_df.columns = ["Line ID", "Nombre de lignes"]
display(line_df)


# ----------- Répartition site + ligne -----------

site_line_df = df_iot.groupby(["site", "line_id"]).size().reset_index(name="Nombre lignes")
display(site_line_df)

,timestamp,site,line_id,sensor_id,temperature_c,vibration_mms,debit_uh
0,2026-04-14T19:21:43,Lyon,1,SLYO-L1-T01,77.92,5.539793,101.27
1,2026-04-27T02:47:12,Lyon,1,SLYO-L1-T01,70.58,3.361715,110.19
2,2026-04-13T18:18:50,Saint-Etienne,1,SSAI-L1-T01,62.37,4.019277,111.28
3,2026-04-05T10:34:03,Roubaix,2,SROU-L2-T01,66.17,4.922531,123.93
4,2026-04-20T10:18:07,Saint-Etienne,3,SSAI-L3-T01,55.56,1.643043,101.40


,line_id,temperature_c,vibration_mms,debit_uh
count,51000.000000,51000.000000,50251.000000,51000.000000
mean,2.005882,73.717034,4.831502,110.050188
std,1.033472,27.005577,2.685963,11.832749
min,1.000000,26.470000,0.000000,80.000000
25%,1.000000,60.237500,3.302777,102.000000
50%,2.000000,66.090000,4.187726,110.100000
75%,3.000000,72.840000,5.171534,118.050000
max,4.000000,160.000000,12.000000,150.000000


,Metric,Valeur
0,Volume lignes,51000
1,Nombre colonnes,7
2,Période début,2026-04-01T00:00:38
3,Période fin,2026-04-29T23:57:53


,Colonne,Nb valeurs manquantes
0,timestamp,0
1,site,0
2,line_id,0
3,sensor_id,0
4,temperature_c,0
5,vibration_mms,749
6,debit_uh,0


,Site,Nombre de lignes
0,Roubaix,20570
1,Saint-Etienne,20248
2,Lyon,10182


,Line ID,Nombre de lignes
0,1,21980
1,3,12068
2,2,11846
3,4,5106


,site,line_id,Nombre lignes
0,Lyon,1,10182
1,Roubaix,1,5038
2,Roubaix,2,5174
3,Roubaix,3,5252
4,Roubaix,4,5106
5,Saint-Etienne,1,6760
6,Saint-Etienne,2,6672
7,Saint-Etienne,3,6816


> **Observations** :
>
> - Volume : 51 000 lignes × 7 colonnes, environ 2,7 MB en mémoire → volume modéré, facilement exploitable.
> - Période : correspond à un historique sur une période continue (probablement ~1 mois).
> - Qualité observée :
Source globalement propre et structurée.
3 types numériques principaux (température, vibration, débit) cohérents avec l’usage industriel.
Présence de valeurs manquantes sur vibration_mms :
→ 749 valeurs manquantes (~1,47 % du dataset).
Colonne timestamp en type object → nécessitera conversion en datetime.
Pas d’anomalie bloquante visible à ce stade.
Données riches et directement exploitables pour analyse machine
> - Risques RGPD : Faible : aucune donnée personnelle directe (pas de nom, email, identifiant RH).
Risque indirect possible par croisement :
combinaison site + line_id + timestamp peut permettre d’inférer quel opérateur était en poste si on croise avec planning RH.
Risque à surveiller mais non bloquant.
> - Pertinence métier : Très élevée
Données directement liées aux conditions physiques de production (température, vibration, débit).
Fort potentiel pour : détecter des dérives machines, identifier des signaux faibles avant défaut qualité.
> - Question pour Sébastien : Les données capteurs IoT sont-elles disponibles et exploitables au moins 24h avant le lancement de la production, ou uniquement pendant le fonctionnement des machines ?

## Source 2 — ERP (JSON)

In [7]:
with (DATA_DIR / "erp_export.json").open() as f:
    orders = json.load(f)
df_erp = pd.DataFrame(orders)
df_erp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   ordre_id         2000 non-null   int64 
 1   produit_ref      2000 non-null   object
 2   site             2000 non-null   object
 3   line_id          2000 non-null   int64 
 4   date_lancement   2000 non-null   object
 5   date_fin_prevue  2000 non-null   object
 6   statut           2000 non-null   object
 7   ouvrier_id       1891 non-null   object
 8   quantite_kg      2000 non-null   int64 
dtypes: int64(3), object(6)
memory usage: 140.8+ KB


In [ ]:
# TODO — explore : value_counts statut, manquants ouvrier_id, etc.

In [8]:
# ----------- Aperçu -----------
display(df_erp.head())

# ----------- Volume -----------
summary_erp = pd.DataFrame({
    "Metric": ["Volume lignes", "Nombre colonnes"],
    "Valeur": [len(df_erp), len(df_erp.columns)]
})
display(summary_erp)

# ----------- Statut des ordres -----------
statut_df = df_erp["statut"].value_counts().reset_index()
statut_df.columns = ["Statut", "Nombre d'ordres"]
display(statut_df)

# ----------- Valeurs manquantes -----------
missing_erp = df_erp.isna().sum().reset_index()
missing_erp.columns = ["Colonne", "Nb valeurs manquantes"]
missing_erp["% manquants"] = (missing_erp["Nb valeurs manquantes"] / len(df_erp)) * 100
display(missing_erp)

# Focus RGPD sur ouvrier_id
ouvrier_missing = pd.DataFrame({
    "Metric": ["Nb valeurs manquantes ouvrier_id", "% manquants"],
    "Valeur": [
        df_erp["ouvrier_id"].isna().sum(),
        (df_erp["ouvrier_id"].isna().sum() / len(df_erp)) * 100
    ]
})
display(ouvrier_missing)

# ----------- Répartition par site -----------
site_erp_df = df_erp["site"].value_counts().reset_index()
site_erp_df.columns = ["Site", "Nombre d'ordres"]
display(site_erp_df)

# ----------- Répartition par ligne -----------
line_erp_df = df_erp["line_id"].value_counts().reset_index()
line_erp_df.columns = ["Line ID", "Nombre d'ordres"]
display(line_erp_df)

# ----------- Quantité produite (indicateur métier) -----------
display(df_erp["quantite_kg"].describe())

,ordre_id,produit_ref,site,line_id,date_lancement,date_fin_prevue,statut,ouvrier_id,quantite_kg
0,100000,ALU-T1-22,Roubaix,3,2026-04-01T22:21:08,2026-04-02T23:21:08,suspendu,EMP-5317,3221
1,100001,INOX-316-4,Saint-Etienne,1,2026-04-26T14:52:52,2026-04-28T15:52:52,termine,EMP-7240,4556
2,100002,ALU-T2-18,Saint-Etienne,3,2026-04-11T09:54:06,2026-04-12T16:54:06,suspendu,EMP-1939,1308
3,100003,ALU-T1-22,Roubaix,1,2026-04-20T22:33:08,2026-04-22T04:33:08,termine,EMP-3531,2968
4,100004,ALU-T2-25,Roubaix,4,2026-04-24T01:03:02,2026-04-25T21:03:02,termine,EMP-8778,3278


,Metric,Valeur
0,Volume lignes,2000
1,Nombre colonnes,9


,Statut,Nombre d'ordres
0,termine,1559
1,en_cours,197
2,suspendu,139
3,annule,105


,Colonne,Nb valeurs manquantes,% manquants
0,ordre_id,0,0.00
1,produit_ref,0,0.00
2,site,0,0.00
3,line_id,0,0.00
4,date_lancement,0,0.00
5,date_fin_prevue,0,0.00
6,statut,0,0.00
7,ouvrier_id,109,5.45
8,quantite_kg,0,0.00


,Metric,Valeur
0,Nb valeurs manquantes ouvrier_id,109.00
1,% manquants,5.45


,Site,Nombre d'ordres
0,Roubaix,1108
1,Saint-Etienne,892


,Line ID,Nombre d'ordres
0,3,575
1,2,573
2,1,549
3,4,303


count    2000.000000
mean     2528.689000
std      1425.269007
min        51.000000
25%      1270.250000
50%      2534.000000
75%      3739.250000
max      4999.000000
Name: quantite_kg, dtype: float64

> **Observations** :
>
> - Volume : 2 000 lignes × 9 colonnes (~140 KB) → volume faible, facilement exploitable.
> - Période : À préciser via date_lancement et date_fin_prevue.
> - Qualité observée :
Données globalement propres et structurées.
Champs métier cohérents : ordre, produit, site, ligne, statut, quantité.
109 valeurs manquantes (~5,45 %) sur ouvrier_id.
Colonnes dates (date_lancement, date_fin_prevue) au format texte → conversion nécessaire.
Répartition des statuts exploitable pour filtrer les ordres (terminés vs en cours).
Aucun défaut bloquant identifié.
> - Risques RGPD : Risque moyen à élevé
Le champ ouvrier_id est un identifiant pseudonymisé → donnée indirectement personnelle.
Risque de réidentification par croisement avec :
logs machines (horodatés),
données IoT,
planning RH.
> - Pertinence métier : Très élevée 
Source clé pour :
contextualiser les données IoT,
relier défauts ↔ ordres de fabrication,
intégrer le planning de production.
anticiper les défauts 24h avant production
→ grâce à date_lancement

## Source 3 — Logs machines (texte)

In [9]:
log_path = DATA_DIR / "logs_machines.log"
n_lines = sum(1 for _ in log_path.open())
print(f"Nombre de lignes : {n_lines:,}")
print(f"Taille fichier : {log_path.stat().st_size / 1024:.1f} Ko")

# Aperçu des 5 premières lignes
with log_path.open() as f:
    for i, line in enumerate(f):
        if i >= 5:
            break
        print(line.rstrip())

Nombre de lignes : 30,000
Taille fichier : 1872.8 Ko
[2026-04-01T00:00:16] Lyon LINE-1 INFO: shift_changed
[2026-04-01T00:01:07] Saint-Etienne LINE-2 INFO: operator_login
[2026-04-01T00:01:34] Saint-Etienne LINE-3 ERROR: vibration_overlimit sensor=SSAI-L3-T01
[2026-04-01T00:04:18] Roubaix LINE-4 INFO: maintenance_completed
[2026-04-01T00:04:35] Lyon LINE-1 INFO: tooling_loaded


In [ ]:
# TODO — compter les types d'événements (INFO/WARN/ERROR), repérer les pics

In [11]:
with log_path.open(encoding="utf-8") as f:
    logs = [line.rstrip() for line in f]

# Création d'un DataFrame simple
df_logs = pd.DataFrame({"raw_log": logs})

# Extraction simple des champs principaux
pattern = r"\[(.*?)\]\s+(\S+)\s+(LINE-\d+)\s+(INFO|WARN|ERROR):\s+(.*)"

df_logs[["timestamp", "site", "line_id", "level", "message"]] = df_logs["raw_log"].str.extract(pattern)

# Conversion timestamp
df_logs["timestamp"] = pd.to_datetime(df_logs["timestamp"], errors="coerce")

# Aperçu
display(df_logs.head())

,raw_log,timestamp,site,line_id,level,message
0,[2026-04-01T00:00:16] Lyon LINE-1 INFO: shift_...,2026-04-01 00:00:16,Lyon,LINE-1,INFO,shift_changed
1,[2026-04-01T00:01:07] Saint-Etienne LINE-2 INF...,2026-04-01 00:01:07,Saint-Etienne,LINE-2,INFO,operator_login
2,[2026-04-01T00:01:34] Saint-Etienne LINE-3 ERR...,2026-04-01 00:01:34,Saint-Etienne,LINE-3,ERROR,vibration_overlimit sensor=SSAI-L3-T01
3,[2026-04-01T00:04:18] Roubaix LINE-4 INFO: mai...,2026-04-01 00:04:18,Roubaix,LINE-4,INFO,maintenance_completed
4,[2026-04-01T00:04:35] Lyon LINE-1 INFO: toolin...,2026-04-01 00:04:35,Lyon,LINE-1,INFO,tooling_loaded


In [13]:
event_level_df = df_logs["level"].value_counts().reset_index()
event_level_df.columns = ["Type événement", "Nombre de lignes"]

display(event_level_df)


,Type événement,Nombre de lignes
0,INFO,22501
1,WARN,5758
2,ERROR,1741


> **Observations** :
>
> - Format : Fichier texte semi-structuré avec une ligne par événement ([timestamp] site LINE-x LEVEL: message).
> - Parsing nécessaire : Oui (indispensable)
Données non directement exploitables sous forme tabulaire
Nécessite extraction via regex ou traitement string
> - Croisement avec les capteurs IoT (corrélation Roubaix line 3 ?) : TTrès pertinent. Les logs peuvent être corrélés avec les données IoT via timestamp, site et line_id. Les événements de type `ERROR` (ex. vibration_overlimit) permettent de confirmer ou d’expliquer les anomalies observées dans les capteurs, notamment sur certaines lignes (ex. Roubaix line 3). Cette complémentarité renforce la capacité du modèle à identifier des situations à risque.

## Synthèse pour `identification_sources.md`

Remplis le tableau d'inventaire dans `../identification_sources.md` à
partir des observations ci-dessus.